### PR TITLE
Introduced TicketBai client.

### PR DESCRIPTION
**Pull request by abdallahbeshi**
## Description

Introduced TicketBai client.

## Type of change
- [ ] Bug fix.
- [x] Feature.
- [ ] Consolidation.

## Checklist
- [ ] Is breaking change.
- [ ] Documentation updated.
- [ ] Tests included (please specify cases).

### CODE REVIEW

File: src/TicketBai/Mews.Fiscalizations.TicketBai/CRC8.cs [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+using Mews.Fiscalizations.Core.Model;
+using Microsoft.AspNetCore.WebUtilities;
+using System.Globalization;
+using System.Text;
+
+namespace Mews.Fiscalizations.TicketBai
+{
+    internal static class CRC8
+    {
+        private static readonly byte[] Crc8Table = new byte[]
```
Comment (abdallahbeshi) [inline src/TicketBai/Mews.Fiscalizations.TicketBai/CRC8.cs:11]: image

File: src/TicketBai/Mews.Fiscalizations.TicketBai/TicketBaiClient.cs [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+using Microsoft.AspNetCore.WebUtilities;
+using System.Globalization;
+
+namespace Mews.Fiscalizations.TicketBai
+{
+    public sealed class TicketBaiClient
+    {
+        private static readonly byte[] Crc8Table = new byte[]
+        {
+            (byte) 0x00, (byte) 0x07, (byte) 0x0E, (byte) 0x09, (byte) 0x1C, (byte) 0x1B, (byte) 0x12, (byte) 0x15,
```
Comment (abdallahbeshi) [inline src/TicketBai/Mews.Fiscalizations.TicketBai/TicketBaiClient.cs:21]: image

File: src/TicketBai/Mews.Fiscalizations.TicketBai/TicketBaiClient.cs [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+            xmlDoc.DocumentElement.AppendChild(xmlDoc.ImportNode(xmlDigitalSignature, true));
+        }
+
+        /// <summary>
+        /// Generates TBAI QR code URI which will be displayed on the invoice.
+        /// The required URI parameters to generate the QR code URI are: TBAI identifier (id) returned in the send invoice response, invoice series (s), invoice number (nf),
+        /// Total amount (i) and the CRC-8 error-detecting-code (cr).
+        /// Example: https://batuz.eus/QRTBAI/?id=TBAI-00000006Y-251019-btFpwP8dcLGAF-237&s=T&nf=27174&i=4.70&cr=007
+        /// </summary>
+        private string GenerateQrCodeUri(string tbaiIdentifier, IOption<String1To20> invoiceSeries, string invoiceNumber, decimal total)
```
Comment (KaliCZ) [inline src/TicketBai/Mews.Fiscalizations.TicketBai/TicketBaiClient.cs:144]: Move the QR stuff into a separate class.

File: src/TicketBai/Mews.Fiscalizations.TicketBai/TicketBaiClient.cs [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+            var xmlDoc = new XmlDocument();
+
+            using (var memStm = new MemoryStream())
+            {
+                var serializer = new XmlSerializer(typeof(Dto.TicketBai));
+                serializer.Serialize(memStm, ticketBaiRequest);
+
+                memStm.Position = 0;
+
+                using (var xtr = new StreamReader(memStm, Encoding.UTF8))
```
Comment (KaliCZ) [inline src/TicketBai/Mews.Fiscalizations.TicketBai/TicketBaiClient.cs:90]: This stuff looks a bit weird. We have to manually set the position of the stream to 0 in order to read it? Is that how other fiscalizations read requests too?

**Review by KaliCZ — APPROVED** [review]

**Pull request merged**